### DISCUSSION TITLE
Is there any way to use the stored value across test files?

### DISCUSSION THREAD

**Discussion by mihirsasmal** [pactumjs/pactum#238] [Q&A]
Hi,

I'm using the `stores()` to store some response values which will be used in different module API call.
When I'm using `$S{variableName}`, it works within same test file multiple tests and api calls. However, it does not resolve when I'm using in another test file.

Is there anyway to use those values across test files?

Thanks,
Mihir

**Comment by ASaiAnudeep** [comment]
Stored values are available across test files in the same run context.

It is not possible if you are running one test file at a time. For example

```shell
mocha test1.js
mocha test2.js
```

**Reply by mihirsasmal** [reply]
I'm running it using Jest and jest config matches the test pattern and run the test. So, I believe it's same run context. 
I have also tried `Jest a.test.ts b.test.ts c.test.ts` This also does not work.

**Reply by ASaiAnudeep** [reply]
Can you provide us a sample code to reproduce the issue.

**Reply by mihirsasmal** [reply]
Here is the [link](https://github.com/mihirsasmal/PactumSamples) to sample code.
You can run tests using this command `npx jest tests/a.test.ts b.test.ts c.test.ts --runInBand` . In this only one will pass and remaining will fail as it uses a stored value.
This command will pass as all the tests in one file `npx jest tests/allCombined.test.ts`

**Reply by mihirsasmal** [reply]
Hi, 
Are you able to reproduce the issue? Is there any way to handle that issue in pactum?

**Reply by ASaiAnudeep** [reply]
Didn't got a chance to reproduce the issue. Will look into it this weekend.

**Reply by ASaiAnudeep** [reply]
Seems the issue is w.r.t jest. Not able to replicate using `mocha`

**Reply by mihirsasmal** [reply]
Ahh.. is it like Jest treats each test file one test and set the environment for that test only and then recreates it for next test?

**Reply by ASaiAnudeep** [reply]
I think so. 

May be it's time to build persistent stores.
